File: 1341-SplitAStringInBalancedStrings/1341-SplitAStringInBalancedStrings.java
// Last updated: 8/5/2025, 2:54:31 PM
class Solution {
    public int balancedStringSplit(String s) {
        int res = 0, count = 0;
        for(char c: s.toCharArray()) {
            count+= c == 'R' ? 1 : -1;
            if(count == 0)
                res++;
        }
        return res;
    }
}

File: 1341-SplitAStringInBalancedStrings/1341-SplitAStringInBalancedStrings.py
# Last updated: 8/5/2025, 2:54:35 PM
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        balance = 0
        charCount = 0
        for i in range(len(s)):
            if s[i] == 'R':
                charCount += 1
            else:
                charCount -= 1
            if charCount == 0:
                balance += 1
        return balance